portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,39 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<String> list = new ArrayList<String>(); 
+    list.add("apples");
+    list.add("oranges");
+    list.add("pineapple");
+
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Lian!</h1>");
+
+    String json = convertToJson(list);
+    response.getWriter().println(json);
+  }
+
+    /**
+   * Converts a ServerStats instance into a JSON string using manual String concatentation.
+   */
+  private String convertToJson(ArrayList<String> list) {
+    String json = "{";
```

Comment:
You're converting to an JSON object here, but you really need to convert to a JSON array.
Arrays will also help you because you'll be able to use loops.
For conversion, Gson will help you, as explained in the tutorial.

Reply:
To follow up on this, in JSON, curly brackets denote an object, and square brackets denote an array. So you want to end up with `[fruit0:'apples', fruit1:'oranges', fruit3:'pineapple']` or, even better, simply an array `['apples', 'oranges', 'pineaple']`.

And yes, definitely use `Gson` to help -- it will turn this whole method into 3 lines.